src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+
+### 8. Can I enable WAL failover during runtime for a single data store cluster?
+
+Yes: If the side disk for WAL failover was configured before the cluster was started, set the [`storage.wal_failover.unhealthy_op_threshold`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-wal-failover-unhealthy-op-threshold) cluster setting to a value less than the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting to enable WAL failover.
+
+No: If the side disk for WAL failover was **not** pre-configured before the cluster was started, changing the value of the [`storage.wal_failover.unhealthy_op_threshold`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-wal-failover-unhealthy-op-threshold) cluster setting to something less than the [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration) cluster setting **will not have any effect and when the disk stalls, WAL failover will not trigger**. In other words, you cannot enable WAL failover at runtime for a single store cluster if you did not [pre-configure a side disk](#single-store-config).
+
+The recommended default value of the [`storage.wal_failover.unhealthy_op_threshold`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-wal-failover-unhealthy-op-threshold) cluster setting is `100ms`.
+
+### 9. Can I change the log buffering on a node before enabling WAL failover?
+
+Yes.
+
+### 10. Does configuring WAL Failover on a running cluster require rolling restarts?
+
+Yes: For a single-store cluster, if you did not have a side disk pre-configured for WAL failover, you will need to do a rolling restart.
+
+Yes: For a multi-store cluster, if you are using file-based logging and did not have log buffering configured according to [3. Log Configuration for WAL Failover](#3-log-configuration-for-wal-failover), you will need to do a rolling restart.
+
+No: In all other cases, you can enable WAL failover without doing a rolling restart.
+
+### 11. After WAL failover is enabled on a cluster, is it safe for the customer to perform all normal operations?
+
+Yes.
+
+### 12. Can I change the value of the `storage.max_sync_duration` cluster setting at cluster runtime?
+
+Yes. To change [`storage.max_sync_duration`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-storage-max-sync-duration), issue a statement like the following:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING storage.max_sync_duration = '${desired_duration}';
+~~~
+
+### 13. If store A has a momentary stall, and it fails over to store B, and B's stall is longer than A's stall, will the WAL failback to A?
```

Comment:
Maybe we should change this question to be:

> When will Cockroach revert to writing a store's WAL on it's primary storage device? What if the secondary storage device stalls?

And the answer something like:

> CockroachDB will monitor the latencies of the primary storage device in the background. As soon as latencies return to acceptable levels, the store will begin writing to the primary device. If the secondary stalls while in use by WAL failover, WAL failover will be unable to limit tail latencies, and the user workload will experience latencies until either the primary or the secondary recovers.

Reply:
updated the question and answer with your text